rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/Diagnostics.java

```diff
@@ -121,14 +185,27 @@ public static Diagnostic translateDiagnostic(IConstructor d, Range range) {
         return result;
     }
 
+    private static Range toRange(ITree t, ColumnMaps cm) {
+        return toRange(TreeAdapter.getLocation(t), cm);
+    }
+
     private static Range toRange(ParseError pe, ColumnMaps cm) {
-        ISourceLocation loc = pe.getLocation();
+        return toRange(pe.getLocation(), cm);
+    }
+
+    private static Range toRange(ISourceLocation loc, ColumnMaps cm) {
         if (loc.getBeginLine() == loc.getEndLine() && loc.getBeginColumn() == loc.getEndColumn()) {
             // zero width parse error is not something LSP likes, so we make it one char wider
-            loc = ValueFactoryFactory.getValueFactory().sourceLocation(loc,
-                loc.getOffset(), loc.getLength() + 1,
-                loc.getBeginLine(), loc.getBeginColumn(),
-                loc.getEndLine(), loc.getEndColumn() + 1);
+            try {
+                loc = ValueFactoryFactory.getValueFactory().sourceLocation(loc,
+                    loc.getOffset(), loc.getLength() + 1,
+                    loc.getBeginLine(), loc.getBeginColumn(),
+                    loc.getEndLine(), loc.getEndColumn() + 1);
+            } catch (Throwable t) {
+                logger.trace("Cannot extend 0-width location for parse error: " + t.getMessage());
```

Comment:
Why do we catch `Throwable`? is there not a more specific exception to catch? In general we shouldn't be catching `Throwable` since that also catches `Error`. I think `Exception` is the most generic we should catch, but even then, is there no more specific one?

nit: log messages should use `"{}"` interpolation instead of string concat. Also there is an overload for log4j2 that allows you just pass in the whole exception I think.